spyder_kernels/customize/spyderpdb.py

```diff
@@ -821,6 +845,11 @@ def enter_recursive_debugger(self, code, filename,
         self.lastcmd = debugger.lastcmd
         get_ipython().pdb_session = self
 
+        # Notify spyder about changed frame. Reset _previous_step to force
+        # notification.
+        self._previous_step = None
+        self.publish_pdb_state()
```

Comment:
I think this is not needed because `postcmd` will be called after this function returns. The line above (`self._previous_step = None`) should be sufficient to address the issue. Am I missing something?

Reply:
No, you’re not. Resetting `self._previous_step` is sufficient.